### PR TITLE
Broken infowindows using Layers endpoint v1.1

### LINKED ISSUE #3648: Broken infowindows using Layers endpoint v1.1
- Create custom infowindow
- Add new layer
- Change order of layers (the layer with the infowindow is the first one)
- Missing infowindows

bugazo

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #3648 
Remaining migration will be finished at #3674, the scope of this is simply fix v1_1 GET call containing proper infowindows (and tooltips)

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3727/))

**Comment by Kartones** [comment]
@juanignaciosl CR please, is only pending manual testing (in local works fine) and merging your branch that also touches the Presenter (some changes are replicated here)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3729/))

File: app/controllers/carto/api/layer_presenter.rb [hunk 1]
```diff
@@ -6,22 +6,81 @@ class LayerPresenter
...
+        @owner_user       = owner_user
+        @options          = options
+        @configuration    = configuration
+        @decoration_data  = decoration_data
       end
 
       def to_poro
-         public_values(@layer).merge(children_for(@layer))
+         poro = public_values(layer).merge(children_for(layer))
+         if options[:viewer_user] and poro['options'] and poro['options']['table_name']
```
Comment (juanignaciosl) [inline app/controllers/carto/api/layer_presenter.rb]: Are we using both `current_viewer` attribute and `options[:viewer_user]`?

**Comment by juanignaciosl** [comment]
Other than the comment, LGTM. Make sure you've tested this both with owner, "other" and anonymous users, since current viewer might be tricky.

File: app/controllers/carto/api/layer_presenter.rb (hunk 1 above)
Comment (Kartones) [inline app/controllers/carto/api/layer_presenter.rb]: Will be fixed at #3674

**Pull request merged**